content/docs/getting-started/hello-world.md

```diff
@@ -1,150 +1,154 @@
-
-stream.write_all(b"hello world\n").await.unwrap();
-
-println!("wrote to stream");
-# }
+```toml
+[dependencies]
+tokio = { version = "0.2", features = ["full"] }
 ```
 
-The [`write_all`] function is implemented for all "stream" like types. It is
-provided by the [`AsyncWriteExt`] trait. Again, the function is asynchronous, so
-no work is done unless `.await` is called. We call `.await` to perform the
-write.
+Then we'll replace `main.rs` with our "hello world" code:
 
-You can find the full example [here][full-code].
+```rust,no_run
+# #![deny(deprecated)]
 
-# Running the code
+use tokio::net::TcpStream;
+use tokio::prelude::*;
 
-[Netcat] is a tool for quickly creating TCP sockets from the command line. The following
-command starts a listening TCP socket on the previously specified port.
+#[tokio::main]
+async fn main() {
+    let mut stream = TcpStream::connect("127.0.0.1:6142").await.unwrap();
+    println!("created stream");
 
-```bash
-$ nc -l 6142
+    let result = stream.write(b"hello world\n").await;
+    println!("wrote to stream; success={:?}", result.is_ok());
+}
 ```
-> The command above is used with the GNU version of netcat that comes stock on many
-> unix based operating systems. The following command can be used with the
-> [NMap.org][NMap.org] version: `$ ncat -l 6142`
 
-In a different terminal we'll run our project.
+The `#[tokio::main]` macro provides common boilerplate for setting up the
+tokio runtime, so that we can write `main()` as an [`async function`]. This
+enables us to call asynchronous functions and write sequential code as if
+they were blocking by using the Rust `await` keyword.
 
-```bash
-$ cargo run
+# Running our app
+
+Our app connects to a socket over TCP and writes to it. Let's run the socat
+command again to test our app:
+
+```
+socat TCP-LISTEN:6142 stdout
 ```
 
-If everything goes well, you should see `hello world` printed from Netcat.
+Now when we run our app with `cargo run` (which will need to be in a different
+window from where we are running socat), we should see socat print
+`hello world`.
 
 # Next steps
 
-We've only dipped our toes into Tokio and its asynchronous model. The next page in
-the guide will start digging a bit deeper into Futures and the Tokio runtime model.
-
-[`io`]: https://docs.rs/tokio/0.2/tokio/io/index.html
-[`net`]: https://docs.rs/tokio/0.2/tokio/net/index.html
-[`write_all`]: https://docs.rs/tokio/0.2/tokio/io/trait.AsyncWriteExt.html#method.write_all
-[`AsyncWriteExt`]: https://docs.rs/tokio/0.2/tokio/io/trait.AsyncWriteExt.html
-[full-code]: https://github.com/tokio-rs/tokio/blob/master/examples/hello_world.rs
-[Netcat]: http://netcat.sourceforge.net/
-[Nmap.org]: https://nmap.org
+Now that we have successfully built a tiny client to get a feel for network
+programming with Tokio, we'll dive into more detail about how everything works.
+
+[Rust programming language]: https://www.rust-lang.org/
+[Tokio 0.1 docs]: https://v0-1--tokio.netlify.com/
```

Comment:
Is this the url we want to use? I was thinking it might go to `https://docs.rs/tokio/0.1.22/tokio/index.html#modules`?

Reply:
I think a pointer to the site will be helpful, then people have the guides too!